src/genpy/generator.py

```diff
@@ -156,9 +156,17 @@ def default_value(msg_context, field_type, default_package):
                 return "b''"
         elif array_len is None: #var-length
             return '[]'
-        else: # fixed-length, fill values
+        else:
+            # fixed-length
             def_val = default_value(msg_context, base_type, default_package)
-            return '[' + def_val + '] * ' + str(array_len)
+            if base_type in [
+                'byte', 'int8', 'int16', 'int32', 'int64', 'uint16', 'uint32',
+                'uint64', 'float32', 'float64', 'string', 'bool'
```

Comment:
looking at http://wiki.ros.org/msg, looks like `byte` should be called `uint8` here.
Or to match how it's defined above in the same file we should add both `char` and `uint8` to this list for completeness